File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/BridgeConferenceParticipantResult.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

/**
 * Structure returned when invoking the bridge conference tax calculation APIs.
 */
public class BridgeConferenceParticipantResult {

    @SerializedName("ParticipantRef")
    private String participantRef = null;

    @SerializedName("ErrorCode")
    private Integer errorCode = null;

    @SerializedName("TransactionType")
    private Integer transactionType = null;

    @SerializedName("ServiceType")
    private Integer serviceType = null;

    @SerializedName("ParticipantTaxes")
    private List<TaxData> participantTaxes = new ArrayList<>();

    /**
     * Sets the participant reference.
     * @param participantRef
     * @return 
     */
    public BridgeConferenceParticipantResult participantRef(String participantRef) {
        this.participantRef = participantRef;
        return this;
    }

    /**
     * Participant reference from request.
     * @return participantRef
     */
    public String getParticipantRef() {
        return participantRef;
    }

    /**
     * Sets the participant reference.
     * @param participantRef 
     */
    public void setParticipantRef(String participantRef) {
        this.participantRef = participantRef;
    }

    /**
     * Sets the error code for participant tax calculation (as applicable).
     * @param errorCode
     * @return 
     */
    public BridgeConferenceParticipantResult errorCode(Integer errorCode) {
        this.errorCode = errorCode;
        return this;
    }

    /**
     * Error code for participant tax calculation (as applicable).
     * @return errorCode
     */
    public Integer getErrorCode() {
        return errorCode;
    }

    /**
     * Sets the error code for participant tax calculation (as applicable).
     * @param errorCode 
     */
    public void setErrorCode(Integer errorCode) {
        this.errorCode = errorCode;
    }

    /**
     * Sets the transaction type used for participant.
     * @param transactionType
     * @return 
     */
    public BridgeConferenceParticipantResult transactionType(Integer transactionType) {
        this.transactionType = transactionType;
        return this;
    }

    /**
     * Transaction type used for participant.
     * @return transactionType
     */
    public Integer getTransactionType() {
        return transactionType;
    }

    /**
     * Sets the transaction type used for participant.
     * @param transactionType 
     */
    public void setTransactionType(Integer transactionType) {
        this.transactionType = transactionType;
    }

    /**
     * Sets the service type used for participant.
     * @param serviceType
     * @return 
     */
    public BridgeConferenceParticipantResult serviceType(Integer serviceType) {
        this.serviceType = serviceType;
        return this;
    }

    /**
     * Service type used for participant.
     * @return serviceType
     */
    public Integer getServiceType() {
        return serviceType;
    }

    /**
     * Sets the service type used for participant.
     * @param serviceType 
     */
    public void setServiceType(Integer serviceType) {
        this.serviceType = serviceType;
    }

    /**
     * Sets the list of objects that contain information about the taxes for an
     * individual participant.
     * @param participantTaxes
     * @return 
     */
    public BridgeConferenceParticipantResult participantTaxes(List<TaxData> participantTaxes) {
        this.participantTaxes = participantTaxes;
        return this;
    }

    /**
     * Adds a tax to the participant result.
     * @param participantTaxesItem
     * @return 
     */
    public BridgeConferenceParticipantResult addParticipantTaxesItem(TaxData participantTaxesItem) {
        this.participantTaxes.add(participantTaxesItem);
        return this;
    }

    /**
     * List of objects that contain information about the taxes for an
     * individual participant.
     * @return participantTaxes
     */
    public List<TaxData> getParticipantTaxes() {
        return participantTaxes;
    }

    /**
     * Sets the list of objects that contain information about the taxes for an
     * individual participant.
     * @param participantTaxes 
     */
    public void setParticipantTaxes(List<TaxData> participantTaxes) {
        this.participantTaxes = participantTaxes;
    }
}

File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/BridgeConferenceParticipantResult.js
/**
 * Structure returned when invoking the bridge conference tax calculation APIs.
 */

function BridgeConferenceParticipantResult() {

    this.ParticipantRef = "";
    this.ErrorCode = 0;
    this.TransactionType = 0;
    this.ServiceType = 0;
    this.ParticipantTaxes = [];

    /**
     * Sets the participant reference.
     * @param participantRef String
     * @return Reference to this object for method chaining. 
     */
    this.participantRef = function (participantRef) {
        this.ParticipantRef = participantRef;
        return this;
    }

    /**
     * Participant reference from request.
     * @return String
     */
    this.getParticipantRef = function () {
        return this.ParticipantRef;
    }

    /* Reference to this object for method chaining.*
     * Sets the participant reference.
     * @param participantRef String 
     */
    this.setParticipantRef = function (participantRef) {
        this.ParticipantRef = participantRef;
    }

    /**
     * Sets the error code for participant tax calculation (as applicable).
     * @param errorCode Number
     * @return Reference to this object for method chaining. 
     */
    this.errorCode = function (errorCode) {
        this.ErrorCode = errorCode;
        return this;
    }

    /**
     * Error code for participant tax calculation (as applicable).
     * @return Number
     */
    this.getErrorCode = function () {
        return this.ErrorCode;
    }

    /* Reference to this object for method chaining.*
     * Sets the error code for participant tax calculation (as applicable).
     * @param errorCode Number 
     */
    this.setErrorCode = function (errorCode) {
        this.ErrorCode = errorCode;
    }

    /**
     * Sets the transaction type used for participant.
     * @param transactionType Number
     * @return Reference to this object for method chaining. 
     */
    this.transactionType = function (transactionType) {
        this.TransactionType = transactionType;
        return this;
    }

    /**
     * Transaction type used for participant.
     * @return Number
     */
    this.getTransactionType = function () {
        return this.TransactionType;
    }

    /* Reference to this object for method chaining.*
     * Sets the transaction type used for participant.
     * @param transactionType Number 
     */
    this.setTransactionType = function (transactionType) {
        this.TransactionType = transactionType;
    }

    /**
     * Sets the service type used for participant.
     * @param serviceType Number
     * @return Reference to this object for method chaining. 
     */
    this.serviceType = function (serviceType) {
        this.ServiceType = serviceType;
        return this;
    }

    /**
     * Service type used for participant.
     * @return Number
     */
    this.getServiceType = function () {
        return this.ServiceType;
    }

    /* Reference to this object for method chaining.*
     * Sets the service type used for participant.
     * @param serviceType Number 
     */
    this.setServiceType = function (serviceType) {
        this.ServiceType = serviceType;
    }

    /**
     * Sets the list of objects that contain information about the taxes for an
     * individual participant.
     * @param participantTaxes Array of TaxData
     * @return Reference to this object for method chaining. 
     */
    this.participantTaxes = function (participantTaxes) {
        this.ParticipantTaxes = participantTaxes;
        return this;
    }

    /**
     * Adds a tax to the participant result.
     * @param participantTaxesItem TaxData object.
     * @return Reference to this object for method chaining. 
     */
    this.addParticipantTaxesItem = function (participantTaxesItem) {
        this.ParticipantTaxes.push(participantTaxesItem);
        return this;
    }

    /**
     * List of objects that contain information about the taxes for an
     * individual participant.
     * @return Array of TaxData
     */
    this.getParticipantTaxes = function () {
        return this.ParticipantTaxes;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of objects that contain information about the taxes for an
     * individual participant.
     * @param participantTaxes Array of TaxData 
     */
    this.setParticipantTaxes = function (participantTaxes) {
        this.ParticipantTaxes = participantTaxes;
    }
}
